Refactor DDP uneven inputs control flags

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

rohan-varma committed:
Refactor DDP uneven inputs control flags

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not. We
basically move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"


This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not. We
basically move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"


This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"

uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"

uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"

uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

pritamdamania87 commented on torch/nn/parallel/distributed.py:
```diff
@@ -90,6 +91,12 @@ def _dump_DDP_relevant_env_vars():
 
 
+
+class DDPUnevenInputsConfig(NamedTuple):
```
Maybe `_DDPUnevenInputsConfig` since this is not a user facing config right?

rohan-varma committed:
Update on "Refactor DDP uneven inputs control flags"

uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**
uneven inputs.
* **#47394 Refactor DDP uneven inputs control flags**

This is a preliminary refactor for the next diff that will add an
additional flag to control whether we throw a StopIteration or not (see https://github.com/pytorch/pytorch/issues/47250 for details). In this change, we move the flags for ddp uneven inputs to a simple class.

Differential Revision: [D24739509](https://our.internmc.facebook.com/intern/diff/D24739509/)

[ghstack-poisoned]

rohan-varma commented:
CI errors are known and due to unrelated flakiness, probably since https://github.com/pytorch/pytorch/pull/46471.
